### PR TITLE
New:

### PR DESCRIPTION
**Pull request by szymongebala**
-returnBorrowedBooks
Validation
-max quantity of borrowed

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -67,6 +67,21 @@
 			<version>1.18.12</version>
 			<scope>provided</scope>
 		</dependency>
+        <dependency>
```
Comment (bederp) [inline pom.xml:70]: 3 times junit import?

File: src/main/java/pl/connectis/projektgrupowy/domain/Book.java [hunk 2]
```diff
@@ -7,12 +7,25 @@
 @Entity
 @Data
 public class Book {
-
+    
+    public Long getId() {
```
Comment (bederp) [inline src/main/java/pl/connectis/projektgrupowy/domain/Book.java]: Do you need getter if there is @Data annotation?

File: src/main/java/pl/connectis/projektgrupowy/domain/Client.java [hunk 3]
```diff
@@ -8,21 +8,44 @@
 @Data
 @Entity
 public class Client {
-
+    
+    public Long getId() {
```
Comment (bederp) [inline src/main/java/pl/connectis/projektgrupowy/domain/Client.java]: Same as above

File: src/main/java/pl/connectis/projektgrupowy/service/ClientServiceImpl.java [hunk 4]
```diff
@@ -29,34 +29,55 @@ public ClientServiceImpl(BookRepository bookRepository, ClientRepository clientR
     public Book borrowBook(Long bookId, Long clientId) {
         Optional<Client> optionalClient = clientRepository.findById(clientId);
         Optional<Book> optionalBook = bookRepository.findById(bookId);
-        if (optionalClient.isPresent() && optionalBook.isPresent()) {
+        //added borrow validation
+        if (optionalClient.isPresent() && optionalBook.isPresent()&&optionalClient.get().getBooks().size()<2 ){
```
Comment (bederp) [inline src/main/java/pl/connectis/projektgrupowy/service/ClientServiceImpl.java:36]: What if validation is not met, what is propagated back to client?

File: src/main/java/pl/connectis/projektgrupowy/service/ClientServiceImpl.java [hunk 5]
```diff
@@ -29,34 +29,55 @@ public ClientServiceImpl(BookRepository bookRepository, ClientRepository clientR
...
             Client client = optionalClient.get();
             Set<Book> bookSet = client.getBooks();
+            
             bookSet.add(optionalBook.get());
             client.setBooks(bookSet);
             clientRepository.save(client);
             return optionalBook.get();
         }
+        System.out.println("No books for u pilgrim");
```
Comment (bederp) [inline src/main/java/pl/connectis/projektgrupowy/service/ClientServiceImpl.java]: Avoid System.out in production code use Logger

File: src/main/java/pl/connectis/projektgrupowy/service/ClientServiceImpl.java [hunk 6]
```diff
@@ -29,34 +29,55 @@ public ClientServiceImpl(BookRepository bookRepository, ClientRepository clientR
...
+            Client client=optionalClient.get();
+            return client.getBooks();
+        }
+        
+        else
+        System.out.println("No books");
+        
+        
+        
         return null;
```
Comment (bederp) [inline src/main/java/pl/connectis/projektgrupowy/service/ClientServiceImpl.java]: if client exists return empty collection, if doesn't throw exception and return appropriate http.status to client

**Review by bederp — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (bederp) [reply pom.xml:70]: Also junit is already provided in spring-boot starter, why are you adding this?

Comment (szymongebala) [reply pom.xml:70]: removed

File: src/main/java/pl/connectis/projektgrupowy/domain/Book.java (hunk 2 above)
Comment (szymongebala) [reply src/main/java/pl/connectis/projektgrupowy/domain/Book.java]: removed

File: src/main/java/pl/connectis/projektgrupowy/domain/Client.java (hunk 3 above)
Comment (szymongebala) [reply src/main/java/pl/connectis/projektgrupowy/domain/Client.java]: removed

File: src/main/java/pl/connectis/projektgrupowy/service/ClientServiceImpl.java (hunk 4 above)
Comment (szymongebala) [reply src/main/java/pl/connectis/projektgrupowy/service/ClientServiceImpl.java:36]: expection added

File: src/main/java/pl/connectis/projektgrupowy/service/ClientServiceImpl.java (hunk 5 above)
Comment (szymongebala) [reply src/main/java/pl/connectis/projektgrupowy/service/ClientServiceImpl.java]: expection added

File: src/main/java/pl/connectis/projektgrupowy/service/ClientServiceImpl.java (hunk 6 above)
Comment (szymongebala) [reply src/main/java/pl/connectis/projektgrupowy/service/ClientServiceImpl.java]: added

**Review by bederp — APPROVED** [review]

**Pull request merged**